service/worker/stream.go

```diff
@@ -189,15 +183,31 @@ func (ps *PushStream) ack(ctx context.Context, message *metrov1.ReceivedMessage)
 // NewPushStream return a push stream obj which is used for push subscriptions
 func NewPushStream(ctx context.Context, nodeID string, subName string, subscriptionCore subscription.ICore, subscriberCore subscriber.ICore, config *HTTPClientConfig) *PushStream {
 	pushCtx, cancelFunc := context.WithCancel(ctx)
+
+	// get subscription Model details
+	subModel, err := subscriptionCore.Get(pushCtx, subName)
+	if err != nil {
+		logger.Ctx(pushCtx).Errorf("error fetching subscription: %s", err.Error())
+		return nil
+	}
+
+	// set http connection timeout from the subscription
+	if subModel.AckDeadlineSec != 0 {
+		// make sure to convert sec to milli-sec
+		config.ConnectTimeoutMS = int(subModel.AckDeadlineSec) * 1e3
+	}
+	httpclient := NewHTTPClientWithConfig(config)
+
 	return &PushStream{
 		ctx:              pushCtx,
 		cancelFunc:       cancelFunc,
 		nodeID:           nodeID,
+		subscription:     subModel,
```

Comment:
if we are passing model, we can remove the subcriptionName attribute below

Reply:
Was keeping it to keep the file diff small :) Anyways did the change now.